docs/design/proposals/carm/cross-account-resource-management.md

```diff
@@ -0,0 +1,177 @@
+- If none of the two annotations are provided ACK will try to find a region from this sources: (see [cli user guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-precedence)): Controller flags, Pod IRSA environment variables...
+
+##### Storing AWS Role ARNs
+
+AccountIDs and their associate AWS Role ARNs will be stored in a `ConfigMap` called "ack-carm-config-map". e.g:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ack-carm-config-map
+  namespace: default
+data:
+  accounts: |
+    123456789012=arn:aws:iam::123456789012:root
+    454545454545=arn:aws:iam::454545454545:role/S3Access
+```
+
+##### IAM Account pivot
+
+When a matching namespace is found, ACK service controllers will search for the IAM Role ARN associated with the AccountID in the `ack-carm-config-map` ConfigMap, and pivot the session to create resources in a new AWS account.
+
+The pivot is done by using the [aws-sdk-go](https://github.com/aws/aws-sdk-go) `session` package. `session.NewSession()` function will load the AWS related environment variables (injected by Pod IRSA) and initialize the [token provider](https://github.com/aws/aws-sdk-go/blob/master/aws/session/session.go#L200-L218) function. Later the token provider function is executed when [`AssumeRoleProvider.Retrieve`](https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/stscreds/assume_role_provider.go#L329) is called and [return a set of credentials](https://github.com/aws/aws-sdk-go/blob/master/aws/credentials/stscreds/assume_role_provider.go#L357-L362) to make pivoted api calls.
+
+##### sequence diagram
+
+![sequence-diagram](./images/carm-sequence-diagram.png)
+
+##### AWS Pod IRSA
+
+ACK Service controllers will be deployed using [AWS Pod IRSA (IAM Roles for service accounts)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html), which will associate a ServiceAccount to the controller pods and result in limiting their permissions when calling the AWS API.
+Pod IRSA will limit ACK controllers API calls to the AWS service resources they are responsible for managing and provide audibility throught AWS CloudTrail.
+The injected IAM Role will be used to pivot the execution with a different target IAM Role using the STS::AssumeRole call.
+In addition to the injected AWS Credentials, Pod IRSA will also add `AWS_REGION` and `AWS_DEFAULT_REGION` to the controllers environment variables. When creating new resources and in case the default region isn't provided via annotations, the controllers will use these environment variables to set a default region for the AWS resources.
+
+##### Auditability (Service controllers side)
+
+ACK service controllers reconcile loops will log every session account pivot using zap logger values e.g:
+
+```golang
+func (r *reconciler) reconcile(req ctrlrt.Request) error {
+  ...
+
+  acctID := r.getOwnerAccountID(res)
+  r.log.WithValues(
+    "account_id", acctID,
```

Comment:
what about the IAM Role ARN used in creating the aws-sdk-go client Session? Should we log that as well? :)